script/lib/omaha.py

```diff
@@ -83,63 +96,72 @@ def get_appguid(channel, platform):
 
 
 def get_app_info(appinfo, args):
-    """
+    '''
     Returns a dict with all the info about the omaha app that we will need
     to perform the upload
-    """
+    '''
 
-    changelog_url = "https://github.com/brave/brave-browser/blob/master/CHANGELOG_DESKTOP.md"
-    chrome_major = get_chrome_version().split('.')[0]
-    chrome_minor = get_chrome_version().split('.')[1]
+    changelog_url = 'https://github.com/brave/brave-browser/blob/master/CHANGELOG_DESKTOP.md'
+
+    if args.version:
+        version_values = args.version.split('.')
+        chrome_major = version_values[0]
+        brave_version = '.'.join(version_values[1:3])
+        version = args.version
+    else:
+        chrome_major = get_chrome_version().split('.')[0]
+        brave_version = get_upload_version()
+        version = chrome_major + '.' + brave_version
+        version_values = version.split('.')
 
     # The Sparkle CFBundleVersion is no longer tied to the Chrome version,
     # instead we derive it from the package.json['version'] string. The 2nd
     # digit is adjusted, and then we utilize that combined with the 3rd digit as
     # the CFBundleVersion. (This is also used in build/mac/tweak_info_plist.py)
-    version_values = get_upload_version().split('.')
-    if int(version_values[0]) >= 1:
-        adjusted_minor = int(version_values[1]) + (100 * int(version_values[0]))
+    if int(version_values[1]) >= 1:
+        adjusted_minor = int(version_values[2]) + (100 * int(version_values[1]))
     else:
         # Fall back to returning the actual minor value
```

Comment:
I wonder how useful this fallback is now. It was intended to be potentially used during our transition to v1.0 but may no longer be necessary.

Reply:
planning some further refactoring of this so will tackle later